### PR TITLE
tests: Replaces dnsutils image used with agnhost (part 4)

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind feature

/sig testing
/sig windows
/area conformance

**What this PR does / why we need it**:

Quite a few images are only used a few times in a few tests. Thus, the images are being centralized into the agnhost image, reducing the number of images that have to be pulled and used.

This PR replaces the usage of the following images with agnhost:

- dnsutils

dnsmasq is a Linux specific binary. In order for the tests to also pass on Windows, CoreDNS should be used instead.

**Which issue(s) this PR fixes**:

Related: #76342

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by smourapina** [comment]
Hello @PatrickLang, @claudiubelu!
This is Bug Triage for release 1.18 checking in, with a friendly reminder that code freeze is coming in about 1 week (5 March). Is this PR still intended for milestone 1.18?

**Comment by claudiubelu** [comment]
> 
> 
> Hello @PatrickLang, @claudiubelu!
> This is Bug Triage for release 1.18 checking in, with a friendly reminder that code freeze is coming in about 1 week (5 March). Is this PR still intended for milestone 1.18?

It is. The original PR was split into 3, and one part already merged. This should too.

**Comment by markyjackson-taulia** [comment]
@listx gentle ping. Code freeze is this Thursday

File: test/utils/image/manifest.go [hunk 1]
```diff
@@ -69,6 +70,7 @@ func initReg() RegistryList {
 		DockerLibraryRegistry:   "docker.io/library",
 		DockerGluster:           "docker.io/gluster",
 		E2eRegistry:             "gcr.io/kubernetes-e2e-test-images",
+		PromoterE2eRegistry:     "us.gcr.io/k8s-artifacts-prod/e2e-test-images",
```
Comment (listx) [inline test/utils/image/manifest.go]: After the [domain flip](https://groups.google.com/d/msg/kubernetes-sig-release/ew-k9PEBckQ/T7dFepHdCAAJ), this should instead be k8s.gcr.io/k8s-artifacts-prod/e2e-test-images.

**Review by listx — COMMENTED** [review]
QQ: Where do these tests run?

File: test/utils/image/manifest.go (hunk 1 above)
Comment (claudiubelu) [reply test/utils/image/manifest.go]: Done

Comment (claudiubelu) [reply test/utils/image/manifest.go]: Done

**Review by claudiubelu — COMMENTED** [review]
> 
> 
> QQ: Where do these tests run?

Let's see...

The test ``should support configurable pod resolv.conf`` is being run by:
    pull-kubernetes-e2e-gce
    pull-kubernetes-e2e-kind
    pull-kubernetes-e2e-kind-ipv6

As you can see, the test passed, there would have been an error otherwise.

As for the other tests, as far as I can tell, they are not presubmit:

``DNS configMap nameserver [IPv4]``
    gci-gce-serial-kube-dns-nodecache
    gci-gce-serial-kube-dns
    gci-gce-serial


``test/e2e/windows/dns.go: should support configurable pod DNS servers``
    aks-engine-azure-windows-master
    most windows jobs.

File: test/utils/image/manifest.go (hunk 1 above)
Comment (johnbelamaric) [reply test/utils/image/manifest.go]: Oh, I think that's not going to work. The flip doesn't happen until April 1!

Comment (claudiubelu) [reply test/utils/image/manifest.go]: Oups, I shouldn't have changed that then. Changing it back and leaving a comment.

**Comment by listx** [comment]
> > QQ: Where do these tests run?
> 
> Let's see...
> 
> The test `should support configurable pod resolv.conf` is being run by:
> pull-kubernetes-e2e-gce
> pull-kubernetes-e2e-kind
> pull-kubernetes-e2e-kind-ipv6
> 
> As you can see, the test passed, there would have been an error otherwise.

Yes correct, thank you for confirming. I don't know why the tests were re-triggered some moments ago (bad Prow regexp?) but all of the presubmits were passing.

> 
> As for the other tests, as far as I can tell, they are not presubmit:
> 
> `DNS configMap nameserver [IPv4]`
> gci-gce-serial-kube-dns-nodecache
> gci-gce-serial-kube-dns
> gci-gce-serial
> 
> `test/e2e/windows/dns.go: should support configurable pod DNS servers`
> aks-engine-azure-windows-master
> most windows jobs.

I suppose the only way to test these jobs is to first merge this PR?

**Comment by listx** [comment]
FTR some of the jobs are postsubmit, e.g. https://testgrid.k8s.io/sig-windows#aks-engine-azure-windows-master. I think having all of the presubmits passing is a good enough bar.

I will wait for the presubmits to go green again.

**Comment by claudiubelu** [comment]
FWIW, the Windows test is passing: https://paste.ubuntu.com/p/dfMT2pVZxM/

**Comment by claudiubelu** [comment]
I've also run the IPv4 tests on a Linux node, and they're passing: https://paste.ubuntu.com/p/Zr6B8JyPMH/
I don't have a IPv6 env to run the IPv6 tests though.

**Comment by listx** [comment]
Maybe the failure for pull-kubernetes-conformance-kind-ipv6-parallel is not a flake?

https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88249/pull-kubernetes-conformance-kind-ipv6-parallel/1235293364576849920/

```
test/e2e/framework/framework.go:703
Mar  4 20:15:08.059: Timed out after 240.019s.
Expected
    <string>: 
to contain substring
    <string>: Error reading file /etc/projected-configmap-volumes/create/data-1
test/e2e/common/projected_configmap.go:370
```

**Comment by claudiubelu** [comment]
> 
> 
> Maybe the failure for pull-kubernetes-conformance-kind-ipv6-parallel is not a flake?
> 
> https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88249/pull-kubernetes-conformance-kind-ipv6-parallel/1235293364576849920/
> 
> ```
> test/e2e/framework/framework.go:703
> Mar  4 20:15:08.059: Timed out after 240.019s.
> Expected
>     <string>: 
> to contain substring
>     <string>: Error reading file /etc/projected-configmap-volumes/create/data-1
> test/e2e/common/projected_configmap.go:370
> ```

Weird, but it's unrelated to the tests updated in this PR. The test that failed is: ``[sig-storage] Projected configMap optional updates should be reflected in volume [NodeConformance] [Conformance]``. Also, the test is using ``mounttest:1.0``, not ``agnhost``.

**Comment by aojea** [comment]
> > Maybe the failure for pull-kubernetes-conformance-kind-ipv6-parallel is not a flake?
> > https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88249/pull-kubernetes-conformance-kind-ipv6-parallel/1235293364576849920/
> > ```
> > test/e2e/framework/framework.go:703
> > Mar  4 20:15:08.059: Timed out after 240.019s.
> > Expected
> >     <string>: 
> > to contain substring
> >     <string>: Error reading file /etc/projected-configmap-volumes/create/data-1
> > test/e2e/common/projected_configmap.go:370
> > ```
> 
> Weird, but it's unrelated to the tests updated in this PR. The test that failed is: `[sig-storage] Projected configMap optional updates should be reflected in volume [NodeConformance] [Conformance]`. Also, the test is using `mounttest:1.0`, not `agnhost`.

this test passed pull-kubernetes-e2e-kind-ipv6 , and `pull-kubernetes-conformance-kind-ipv6-parallel` is a subset of it ... it should be unrelated but feel free to ping me if you find something odd with IPv6

/test pull-kubernetes-conformance-kind-ipv6-parallel

**Comment by listx** [comment]
Please remove hold when the failing test situation has been worked out.

/hold
/approve
/lgtm

**Comment by markyjackson-taulia** [comment]
@claudiubelu the test passed.

**Pull request merged**